### PR TITLE
Spice rotation update to use ALE

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updates SpiceRotation class in ISIS to use ALE.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/cassini/objs/VimsCamera/VimsCamera.truth [hunk 1]
```diff
@@ -141,8 +141,8 @@ DeltaSample = 0.000000000
 DeltaLine = 0.000000000
 
 For center pixel position ...
-Latitude off by: 0.0229595838205015
-Longitude off by: 0.0098872441018614
+Latitude off by: 0.0229595838120105
+Longitude off by: 0.0098872441060109
```
Comment (jessemapel) [inline isis/src/cassini/objs/VimsCamera/VimsCamera.truth]: For these unit tests you should updated the expected latitude and longitude values, not how much they are off by.

File: isis/src/hayabusa/objs/HayabusaNirsCamera/HayabusaNirsCamera.truth [hunk 2]
```diff
@@ -30,5 +30,5 @@ DeltaSample = 0.000000000
 DeltaLine = 0.000000000
 
 For center pixel position ...
-Latitude OK
-Longitude OK
+Latitude off by: 0.0000000017899069
+Longitude off by: 0.0000000040511168
```
Comment (jessemapel) [inline isis/src/hayabusa/objs/HayabusaNirsCamera/HayabusaNirsCamera.truth]: Same as vims test

File: isis/src/newhorizons/objs/NewHorizonsLeisaCamera/NewHorizonsLeisaCamera.truth [hunk 3]
```diff
@@ -4367,7 +4367,7 @@ Test forward and backward line/samp to lat/lon delta for Band #256
 
 For center pixel position ...
 Latitude OK
-Longitude OK
+Longitude off by: -0.0000000000004299
```
Comment (jessemapel) [inline isis/src/newhorizons/objs/NewHorizonsLeisaCamera/NewHorizonsLeisaCamera.truth]: Same as vims test

File: isis/src/base/objs/SpiceRotation/SpiceRotation.h [hunk 4]
```diff
@@ -426,7 +424,7 @@ namespace Isis {
       void checkForBinaryPck();
 
       int cacheSize() {
-        return p_cache.size();
+        return m_orientation->getRotations().size();
```
Comment (jessemapel) [inline isis/src/base/objs/SpiceRotation/SpiceRotation.h]: This needs to check that the m_orientations pointer is valid. If it's invalid, then it should return 0.

File: isis/src/base/objs/SpiceRotation/SpiceRotation.cpp [hunk 5]
```diff
@@ -181,13 +181,19 @@ namespace Isis {
     p_hasAngularVelocity = rotToCopy.p_hasAngularVelocity;
     m_frameType = rotToCopy.m_frameType;
 
+    m_orientation = new ale::Orientations;
+    *m_orientation = *rotToCopy.m_orientation;
```
Comment (jessemapel) [inline isis/src/base/objs/SpiceRotation/SpiceRotation.cpp]: This will seg fault if rotToCopy.m_orientation is null

**Review by jessemapel — COMMENTED** [review]
A couple small comments in SpiceRotation, mostly about pointers. For the camera tests, don't update the truth with the "off by ...." stuff, change the expected value in the unitTest.cpp files. To get the new value set them to 0.0 and then copy and paste the output back into it.

File: isis/src/base/objs/SpiceRotation/SpiceRotation.cpp (hunk 5 above)
Comment (krlberry) [reply isis/src/base/objs/SpiceRotation/SpiceRotation.cpp]: Thanks for catching this!

File: isis/src/base/objs/SpiceRotation/SpiceRotation.h (hunk 4 above)
Comment (krlberry) [reply isis/src/base/objs/SpiceRotation/SpiceRotation.h]: Thanks for catching this!

**Comment by krlberry** [comment]
@jessemapel Thanks for the comments! Some of the unitTests were already outputting lat/lon offsets instead of OK/OK, but I agree that they _should_ be outputting OKs, so I'll update accordingly.

**Review by jessemapel — DISMISSED** [review]

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
This is good to merge once the data's been checked in

**Comment by krlberry** [comment]
Test data is all checked in!

**Pull request merged**